keps/sig-cloud-provider/2025-extend-konnectivity-for-both-directions/README.md

```diff
@@ -0,0 +1,402 @@
+
+### Graduation Criteria
+
+
+### Upgrade / Downgrade Strategy
+
+As we cannot rely on Kubernetes resources like DaemonSet or Deployment (see [Deployment Model](#deployment-model)), upgrading the agent will be a similar process as upgrading the Kubelet. This can be achieved either in place, by updating and restarting the systemd service on every node or by changing the manifests if static pods are used. The alternative, in case in-place updates are not an option, is to create a new set of nodes with the new configuration and replace the old ones.
+
+### Version Skew Strategy
+
+
+## Production Readiness Review Questionnaire
+
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [*] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: EnableClusterToMasterTraffic
+    - Components depending on the feature gate: Konnectivity server and agent
+  - [*] Other
+    - This is the procedure to enable in a live cluster assuming that the
+      secure port of KAS is `6443`, the ip address chosen for the agent is
+      `10.0.0.1` and the Konnectivity server is deployed as a DaemonSet on all
+      master nodes:
+      1. Deploy the Konnectivity agent in the nodes by using a static pod or a
+         systemd unit. Add the following flags to enable the feature and to
+         forward local port `6443` to `localhost:6443` on Konnectivity server
+         side:
+         `--feature-gates="EnableClusterToMasterTraffic=true"`
+         `--target=6443:localhost:6443`
+      2. Deploy the Konnectivity server on master nodes following the
+         documented procedure, and adding the following flags to enable the
+         feature and to allow traffic to `localhost:6443`:
+         `--feature-gates="EnableClusterToMasterTraffic=true"`
+         `--allowed-destination=localhost:6443`.
+      3. Add the following flags to the KAS `--advertise-address=10.0.0.1`,
+         `--secure-port=6443`, and restart the process for all master nodes.
+    - In order to disable the feature you just need to remove or re-establish
+      the original `--advertise-address` flag and restart the KAS services on
+      all nodes.
+
+* **Does enabling the feature change any default behavior?**
+  Yes, all traffic originated from node network will pass from the Konnectivity
+  agents. In case network proxies are used, the agent should be configured
+  properly, and the ip used by the agents should be included in `no_proxy`
+  environment variable.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes, it can be disabled by simply changing the KAS `--advertise-address` and
```

Comment:
Seems like you'd need to update the workers to remove the konnectivity agent t hat's configured locally.

Reply:
indeed @deads2k the agents should be removed. I'll add this step